tests/llsd_test.py

```diff
@@ -1977,3 +1977,59 @@ def test_uuid_map_key(self):
         self.assertEqual(llsd.format_notation(llsdmap), b"{'00000000-0000-0000-0000-000000000000':'uuid'}")
 
 
+@unittest.skipIf(PY2, "These tests require Python 3")
```

Comment:
Until we formally drop support for Python 2.7 in this library, we should run as many tests as possible with it, to avoid cases like those discussed above where functionality is broken / not implemented as intended for Python 2.

Ideally we could also split these tests into a commit that precedes the actual fix, to show that these cases already raise exceptions, though with different exception classes & messages in some cases.

Let me know if you're busy and I can take a crack at that, if you like!